### PR TITLE
Migration Bugs

### LINKED ISSUE #1567: Provide clickable migration of file store ZNG to Zed lakes
We expect storage in the new Brim experience to be focused exclusively on Zed lakes. However, users may still have existing "file store" Spaces from the old Brim experience that they may want to continue to work with. At a minimum, we could point users to an article that describes the change, tells them where on the filesystem they'll find the `all.zng` files for their old Spaces, and describes `zapi` command lines that could be used to manually re-import those into Zed lakes. However, when discussing as a group, there was some consensus (@jameskerr in particular seemed on-board) toward some clickable helpers, such that if the user clicks a "file store" Space, we pop up notification regarding the new file format (perhaps link to a Support article that describes it in more detail) and provides a push-button option that would invoke the necessary `zapi post` command line to do it for them, perhaps using the same Space name as the old one with `-lake` appended to it. The user should be able to still right-click and Delete the old Space once they're satisfied with how the data looks in the new Zed lake representation.

In the same pop-up we can also explain that if the logs in the Space were originally created from a pcap, they'll lose the ability to pivot from the logs in the Zed lake back to the pcap slices, so if that's something they want to preserve they'll want to re-import the pcap via Brimcap.

### PR DESCRIPTION
**Pull request by jameskerr**
Closes #1567 

The path to the brimcap exe is now correct for the migration. Previously it was correct in dev, but not in a release.

I've been excited to introduce "ipc service objects" to our electron app and decided this was a good time. There is a new file `app/ipc/paths.ts` that includes a class with a single function `brimcap()`. It returns the path to the brimcap executable. This class then gets transformed into an ipcClient and can be used in the renderer like so.

```
import {pathsClient} from "app/ipc/paths"

const cmd = await pathsClient.brimcap()
```

And it can be used in the main process like so:
```
import {paths} from "app/ipc/paths"
const cmd = paths.brimcap()
```

The main process just needs to "serve" this object to create ipc handlers for all its methods.

```
// main.ts
import {paths} from "app/ipc/paths"

function main() {
  serve(paths)
}
```

I think this is very nice pattern and would love feedback. Maybe we can tweek the api to make it a one-liner to create the client and server. I can think of lots of ipc calls we make that could be converted to little service objects like this.

### CODE REVIEW

File: app/legacy/space-migration/space-migrator.ts [hunk 1]
```diff
@@ -24,19 +29,22 @@ export default class SpaceMigrator {
...
       ])
+
       const linesErr = readline.createInterface({input: this.process.stderr})
       let total = 0
       let count = 0
       let space = ""
       let totalRegexp = /^migrating (\d+) spaces$/
 
       linesErr.on("line", (line) => {
+        console.log(line)
```
Comment (mason-fish) [inline app/legacy/space-migration/space-migrator.ts:47]: Was this just for debug or did you want to keep this in?

**Review by mason-fish — APPROVED** [review]
Overall this lgtm! I'd like to chat more about this new service pattern, which looks v interesting, but I don't want to hold back the migration fix included here. :shipit:

**Comment by philrz** [comment]
@mason-fish: RE: chatting more, I just made a note to tee it up as an "after party" topic at our team sync tomorrow. Particularly with @jameskerr about to be away for a bit, I know other Devs are taking an interest in understanding more of the JavaScript.

File: app/legacy/space-migration/space-migrator.ts (hunk 1 above)
Comment (jameskerr) [reply app/legacy/space-migration/space-migrator.ts:47]: I want that in, in case the user needs more info about a problem.

**Pull request merged**